### DISCUSSION TITLE
Would it be possible to use blatann to conduct legacy DFU updates?

### DISCUSSION THREAD

**Discussion by tjhasan** [ThomasGerstenberg/blatann#81] [Q&A] [answered]
I am interested in using blatann to create a legacy dfu updater. The reason for this is, currently there is no way to conduct legacy DFU updates for nordic devices on desktop since the software that was used for it is now deprecated.

It is still possible to conduct legacy DFU updates via the mobile application, however it is much more convenient to be able to create and upload firmware images on one machine.

Regardless, while looking through the blatann library, it seems that I will be able to replicate the entire legacy dfu process, i.e the data reads and writes, except the actual firmware image upload.

Does blatann have any method of uploading firmware images? If so how would I go about doing this? If what I am seeking already exists, where would I be able to find this solution?

Any help or guidance would be greatly appreciated.

**Comment by ThomasGerstenberg** [comment, marked as answer]
It should definitely be possible. Blatann doesn't have any of the DFU functionality already built in so you'll have to reimplement the process yourself. I'm not familiar with their legacy DFU process, I've only used their secure DFU

**Reply by tjhasan** [reply]
I had a feeling that was the case. I'm currently working on a module to implement the legacy DFU process since a lot of Nordic devices still use legacy as opposed to Secure. I just wanted to double-check and make sure that I wasn't reinventing the wheel. Thank you for the input!
